swap/swap.go

```diff
@@ -457,51 +460,83 @@ func (s *Swap) getContractOwner(ctx context.Context, address common.Address) (co
 	return contr.Issuer(nil)
 }
 
-// StartChequebook deploys a new instance of a chequebook if chequebookAddr is empty, otherwise it wil bind to an existing instance
-func (s *Swap) StartChequebook(chequebookAddr common.Address) error {
-	if chequebookAddr != (common.Address{}) {
-		if err := s.BindToContractAt(chequebookAddr); err != nil {
-			return err
+func promptInitialDepositAmount() (uint64, error) {
+	// need to prompt user for initial deposit amount
+	// if 0, can not cash in cheques
+	prompter := console.Stdin
+
+	// ask user for input
+	input, err := prompter.PromptInput("Please provide the amount in Wei which will deposited to your chequebook upon deployment: ")
+	if err != nil {
+		return 0, err
+	}
+	// check input
+	val, err := strconv.ParseInt(input, 10, 64)
+	if err != nil {
+		// maybe we should provide a fallback here? A bad input results in stopping the boot
+		return 0, fmt.Errorf("Conversion error while reading user input: %v", err)
+	}
+	return uint64(val), nil
+}
+
+// StartChequebook start the chequebook, taking into account the chequebookAddress passed in by the user and the chequebook addresses saved on the node's database
+func (s *Swap) StartChequebook(chequebookAddrFlag common.Address, initialDepositAmount uint64) error {
+	chequebookToUse, err := s.loadChequebook()
+	// error reading from disk
+	if err != nil && err != state.ErrNotFound {
+		return fmt.Errorf("Error reading previously used chequebook: %s", err)
+	}
+	// nothing written to state disk before, no flag provided: deploying new chequebook
+	if err == state.ErrNotFound && chequebookAddrFlag == (common.Address{}) {
+		var toDeposit = initialDepositAmount
+		if toDeposit == 0 {
+			toDeposit, err = promptInitialDepositAmount()
+			if err != nil {
+				return err
+			}
 		}
-		log.Info("Using the provided chequebook", "chequebookAddr", chequebookAddr)
-	} else {
-		if err := s.Deploy(context.Background()); err != nil {
-			return err
+		chequebook, err := s.Deploy(context.TODO(), toDeposit)
+		if err != nil {
+			return fmt.Errorf("Error deploying chequebook: %v", err)
 		}
-		log.Info("New SWAP contract deployed", "contract info", s.DeploySuccess())
+		chequebookToUse = chequebook
 	}
-	return nil
-}
 
-// BindToContractAt binds an instance of an already existing chequebook contract at address and sets chequebookAddr
-func (s *Swap) BindToContractAt(address common.Address) (err error) {
+	// read from state, but provided flag is not the same
+	if err == nil && (chequebookAddrFlag != common.Address{} && chequebookAddrFlag != chequebookToUse) {
+		return fmt.Errorf("Attempting to connect to provided chequebook, but different chequebook used before")
+	}
+	if chequebookAddrFlag != (common.Address{}) {
+		chequebookToUse = chequebookAddrFlag
+	}
+	auditLog.Info("Using the chequebook", "chequebookAddr", chequebookToUse)
```

Comment:
I think this log line is less useful than the previous one informing about a deploy success. It can be additional but it should be clearer. Also always make sure that what is printed is actually readable (what does this print? It may very well be ok, didn't check)

Reply:
Changed to:
`auditLog.Info("Deployed chequebook", "deposit", toDeposit, "owner", owner)`

Reply:
Print is verified